### PR TITLE
Misc cleanup needed for bigdatagenomics/cannoli#65

### LINKED ISSUE #1701: Allow silencing the # of reads in fragment warning in InterleavedFastqInFormatter
If you're realigning a file that was already aligned, [this](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala#L62-L68) warning creates lots of log noise.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1701, #1702, #1703.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala [hunk 1]
```diff
@@ -59,11 +60,11 @@ class InterleavedFASTQInFormatter private (
     iter.flatMap(frag => {
       val reads = converter.convertFragment(frag).toSeq
 
-      if (reads.size < 2) {
+      if (enableLogging && reads.size < 2) {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala]: would simply dropping to `log.info` accomplish the same goal?

Comment (fnothaft) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala]: I mean, it really is a warning message. However, if you're doing development of a piped tool and you want to read your executor logs to figure out why the tool running under the pipe is blowing up, the warnings get in the way of reading the logs.

If you're not 100% sold on this change, I'm OK with dropping it. It was just for my convenience when developing.

Comment (heuermh) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala]: I don't have data to test these particular checks (though I suppose I could invent some) and am not all that confident our logging configuration works.  Could you try adding
```
log4j.logger.org.bdgenomics.adam.rdd.fragment.InterleavedFASTQInFormatter=ERROR
```
to your log4j.properties and see if it does the trick?

Comment (fnothaft) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/InterleavedFASTQInFormatter.scala]: Yeah, agreed. I've rebased this and dropped the commit that resolves #1701.

**Comment by fnothaft** [comment]
This is good to go from my side.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AnySAMOutFormatter.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ import scala.collection.mutable.ListBuffer
  * An OutFormatter that automatically infers whether the piped input is SAM or
  * BAM. Autodetecting streamed CRAM is not currently supported.
  */
-class AnySAMOutFormatter extends OutFormatter[AlignmentRecord] {
+case class AnySAMOutFormatter(stringency: ValidationStringency = ValidationStringency.STRICT) extends OutFormatter[AlignmentRecord] {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AnySAMOutFormatter.scala]: From the Jenkins failures, looks like the default ctr parameter value isn't available to Python.  Could you add a no-arg `apply` method or update the Python code?

**Comment by heuermh** [comment]
Thank you, @fnothaft

**Pull request merged**